test/e2e/minscale_readiness_test.go

```diff
@@ -180,31 +180,36 @@ func privateServiceName(t *testing.T, clients *test.Clients, revisionName string
 	return privateServiceName
 }
 
-func waitForDesiredScale(clients *test.Clients, serviceName string, cond func(int) bool) error {
+// waitForDesiredScale retuns the last observed number of pods and/or error if the cond
+// callback is never satisfied.
+func waitForDesiredScale(clients *test.Clients, serviceName string, cond func(int) bool) (latestReady int, err error) {
 	endpoints := clients.KubeClient.Kube.CoreV1().Endpoints(test.ServingNamespace)
 
-	return wait.PollImmediate(250*time.Millisecond, 1*time.Minute, func() (bool, error) {
+	return latestReady, wait.PollImmediate(250*time.Millisecond, time.Minute, func() (bool, error) {
 		endpoint, err := endpoints.Get(serviceName, metav1.GetOptions{})
 		if err != nil {
 			return false, nil
 		}
-		return cond(resources.ReadyAddressCount(endpoint)), nil
+		latestReady = resources.ReadyAddressCount(endpoint)
+		return cond(latestReady), nil
 	})
 }
 
-func ensureDesiredScale(clients *test.Clients, serviceName string, cond func(int) bool) bool {
+func ensureDesiredScale(clients *test.Clients, serviceName string, cond func(int) bool) (latestReady int, observed bool) {
 	endpoints := clients.KubeClient.Kube.CoreV1().Endpoints(test.ServingNamespace)
 
-	return wait.PollImmediate(250*time.Millisecond, 5*time.Second, func() (bool, error) {
+	err := wait.PollImmediate(250*time.Millisecond, 10*time.Second, func() (bool, error) {
 		endpoint, err := endpoints.Get(serviceName, metav1.GetOptions{})
 		if err != nil {
 			return false, nil
 		}
 
-		if scale := resources.ReadyAddressCount(endpoint); !cond(scale) {
-			return false, fmt.Errorf("scale %d didn't meet condition", scale)
+		if latestReady = resources.ReadyAddressCount(endpoint); !cond(latestReady) {
+			return false, fmt.Errorf("scale %d didn't meet condition", latestReady)
 		}
 
 		return false, nil
-	}) == wait.ErrWaitTimeout
+	})
+
+	return latestReady, err == wait.ErrWaitTimeout
```

Comment:
wonder if we should log the error here, seems like it's getting gobbled and it may be a clue?

Reply:
Yeah, I did this while locally debugging, but then removed, but sure, let's keep it.